STIPS/STIP-009.md

```diff
@@ -0,0 +1,1667 @@
++ require that extension state in delegatedManager's *extensionAllowlist* is *INITIALIZED*
++ call *_initializeModule(_delegatedManager.setToken(), _delegatedManager, _settings)*
+
+----
+
+> initializeExtension
+
+Initialize the StreamingFeeSplitExtension on the DelegatedManager
+
+```solidity
+function initializeExtension(address _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager)
+```
+
++ require that extension state in delegatedManager's *extensionAllowlist* is *PENDING*
++ call *_initializeExtension(_delegatedManager.setToken(), _delegatedManager)*
++ emit *StreamingFeeSplitExtensionInitialized* event
+
+----
+
+> initializeModuleAndExtension
+
+Initializes StreamingFeeSplitExtension to the DelegatedManager and StreamingFeeModule to the SetToken
+
+```solidity
+function initializeModuleAndExtension(
+    IDelegatedManager _delegatedManager,
+    IStreamingFeeModule.FeeState memory _settings
+)
+    external
+    onlyOwnerAndValidManager(_delegatedManager)
+```
+
++ require that extension state in delegatedManager's *extensionAllowlist* is *PENDING*
++ extract SetToken from DelegatedManager *ISetToken setToken = _delegatedManager.setToken()*
++ call *_initializeExtension(setToken, _delegatedManager)*
++ call *_initializeModule(_delegatedManager.setToken(), _delegatedManager, _settings)*
++ emit *StreamingFeeSplitExtensionInitialized* event
+
+----
+
+> removeExtension
+
+Remove an existing SetToken and DelegatedManager tracked by the StreamingFeeSplitExtension
+
+```solidity
+function removeExtension() external override
+```
+
++ Extract DelegatedManager *IDelegatedManager(msg.sender)*
++ Extract SetToken *delegatedManager.setToken()*
++ call *_removeExtension(setToken, delegatedManager*
+
+### Internal functions
+
+> _initializeModule
+
+Internal function to initialize StreamingFeeModule on the SetToken associated with the DelegatedManager
+
+```solidity
+function _initializeModule(
+    ISetToken _setToken,
+    IDelegatedManager _delegatedManager,
+    IStreamingFeeModule.FeeState memory _settings
+)
+    internal
+```
+
++ Formulate call to initialize module from manager
+    ```solidity
+    bytes memory callData = abi.encodeWithSignature(
+        "initialize(address,(address,uint256,uint256,uint256))",
+        _setToken,
+        _settings);
+    );
+    ```
++ call *_invokeManager(_delegatedManager, address(streamingFeeModule), callData)*
+
+## Checkpoint 3
+
+Before we move onto the implementation phase we want to make sure that we are aligned on the spec. All contracts should be specced out, their state and external function signatures should be defined. For more complex contracts, internal function definition is preferred in order to align on proper abstractions. Reviewer should take care to make sure that all stake holders (product, app engineering) have their needs met in this stage.
+
+**Reviewer**:
+
+## Implementation
++ [Final Implementation PR](https://github.com/SetProtocol/set-v2-strategies/pull/23)
```

Comment:
Anyone have strong feelings on this? Should I include intermediate implementation PRs? ([#14](https://github.com/SetProtocol/set-v2-strategies/pull/14), [#15](https://github.com/SetProtocol/set-v2-strategies/pull/15), [#17](https://github.com/SetProtocol/set-v2-strategies/pull/17), [#21](https://github.com/SetProtocol/set-v2-strategies/pull/21), [#23](https://github.com/SetProtocol/set-v2-strategies/pull/23))